### PR TITLE
feat: Support for non-threaded Receive via Update method

### PR DESCRIPTION
**Pull request by eriksvedang**

### CODE REVIEW

File: src/lib/Client.cs [hunk 1]
```diff
@@ -110,15 +111,29 @@ public void Send(byte[] data, IPEndPoint receivedEndpoint)
                 throw new Exception($"UDP Send: Packet too big {data.Length}");
             }
 
-            port.Send(data);
+            port.Send(receivedEndpoint, data);
         }
 
-        byte[] Receive(out IPEndPoint receivedEndpoint)
+        public byte[] Receive(out IPEndPoint receivedEndpoint)
```
Comment (piot) [inline src/lib/Client.cs:117]: Does Receive need to be public?

File: src/lib/Client.cs [hunk 2]
```diff
@@ -110,15 +111,29 @@ public void Send(byte[] data, IPEndPoint receivedEndpoint)
...
         {
             var (data, hostEndpoint) = port.Receive();
-
             receivedEndpoint = hostEndpoint;
             return data;
         }
+
+        private void Relay()
+        {
+            var receivedEndpoint = new IPEndPoint(IPAddress.Any, 32001);
```
Comment (piot) [inline src/lib/Client.cs:126]: receivedEndpoint should not be set to something, just use the `var out`

File: src/lib/Client.cs [hunk 3]
```diff
@@ -33,23 +33,27 @@ public class Client
     {
         IPort port;
         IPEndPoint endpoint;
-        IPacketReceiver receiver;
+        public IPacketReceiver receiver;
```
Comment (piot) [inline src/lib/Client.cs:36]: The `receiver` shouldn't be public?

Comment (eriksvedang) [reply src/lib/Client.cs:36]: Right, was a forgotten hack :)

File: src/lib/Client.cs (hunk 1 above)
Comment (eriksvedang) [reply src/lib/Client.cs:117]: Sorry, same hack.

File: src/lib/Client.cs (hunk 2 above)
Comment (eriksvedang) [reply src/lib/Client.cs:126]: Agreed.

**Comment by piot** [comment]
Thanks for contributing!

**Pull request merged**